### ISSUE TITLE
SQLite doesn't support foreign keys

### ISSUE THREAD

**Issue by amacal** [fluentmigrator/fluentmigrator#356]
I tried to execute following migration

```
        Create.Table("folder")
            .WithColumn("id").AsGuid().NotNullable().PrimaryKey()
            .WithColumn("name").AsString().NotNullable()
            .WithColumn("path").AsString().NotNullable();

        Create.Table("file")
            .WithColumn("id").AsGuid().NotNullable().PrimaryKey()
            .WithColumn("path").AsString().NotNullable().Unique("uq_file_path")
            .WithColumn("folder_id").AsGuid().NotNullable().ForeignKey("fk_file_folder", "folder", "id");
```

and I became following messages

[+] CreateTable folder
CREATE TABLE 'folder' ('id' UNIQUEIDENTIFIER NOT NULL, 'name' TEXT NOT NULL, 'path' TEXT NOT NULL, CONSTRAINT 'PK_folder' PRIMARY KEY ('id'))
=> 0s
[+] CreateTable file
CREATE TABLE 'file' ('id' UNIQUEIDENTIFIER NOT NULL, 'path' TEXT NOT NULL, 'folder_id' UNIQUEIDENTIFIER NOT NULL, CONSTRAINT 'PK_file' PRIMARY KEY ('id'))
=> 0s
[+] CreateIndex file (path)
CREATE UNIQUE INDEX 'uq_file_path' ON 'file' ('path' ASC)
=> 0s
[+] CreateForeignKey fk_file_folder file(folder_id) folder(id)
No SQL statement executed.
=> 0,001s

The foreign key 'fk_file_folder' wasn't added. I read that there is a lot of issue with altering the SQLite table. It isn't simply supported. I didn't want to alter the table. I want to create it with foreign keys. Is it possible to do it in a single step?

**Comment by ha11owed** [comment]
I have implemented the ability to add foreign key to sqlite tables when creating the table.

The only change is in SqliteGenerator.cs (https://github.com/ha11owed/fluentmigrator/blob/master/src/FluentMigrator.Runner/Generators/SQLite/SQLiteGenerator.cs)

I was unable to create a real unit test, but I tested with one of our migrations and seems to work fine:

return Create.Table(tableName)
                .WithColumn("Id").AsInt64().PrimaryKey("PK_" + tableName).Identity()
                .WithColumn("CreatedById").AsInt64().NotNullable().ForeignKey("FK_" + tableName.Replace("_", "") + "_Users_Created", "Users", "Id").OnDeleteOrUpdate(Rule.None)
                .WithColumn("UpdatedById").AsInt64().NotNullable().ForeignKey("FK_" + tableName.Replace("_", "") + "_Users_Updated", "Users", "Id").OnDeleteOrUpdate(Rule.None)
                .WithColumn("CreatedAt").AsDateTime().NotNullable()
                .WithColumn("UpdatedAt").AsDateTime().NotNullable()

If anyone is interested in creating unit tests, I can commit my code, but I can't seem to manage to follow the testing methods used until now.

**Comment by STVM** [comment]
@ha11owed 
I am interested in adding foreign key in sqlite.
Will you commit your modifications to fluentmigrator?
Can I help you in doing unit tests ?

**Comment by ha11owed** [comment]
Hi, yes of course, I will clean the code up a bit and try to get it pushed. The problem with the tests is that they test more or less the internal methods of fluent migrator, and I don't quite know how that maps to the changes I have done

**Comment by fubar-coder** [comment]
Fixed by #638
